docs/index.asciidoc

```diff
@@ -88,6 +88,14 @@ will be similar to events directly indexed by {plugin-uc} into Elasticsearch.
 endif::[]
 
 
+[id="plugins-{type}s-{plugin}-memory"]
+===== Memory usage
+
+This plugin uses direct memory which is not included in the heap memory.
+By default, it is the same as the heap size (Xmx), for example, setting 10GB Xmx means setting 10GB of direct memory.
+Make sure heap and direct memory does not exceed the total size of the server. You can set the amount of direct memory with -XX:MaxDirectMemorySize.
```

Comment:
```suggestion
Make sure heap and direct memory combined does not exceed the total memory available on the server.
You can set the amount of direct memory with `-XX:MaxDirectMemorySize` in {logstash-ref}/jvm-settings.html[Logstash's JVM Settings].
```

We should also add an off-heap section to [Logstash's JVM Settings](https://github.com/elastic/logstash/blob/main/docs/static/config-details.asciidoc).